src/transaction/transaction.rs

```diff
@@ -644,6 +669,10 @@ impl Transaction {
             .retry_region(self.options.retry_options.region_backoff.clone())
             .merge(Collect)
             .plan();
+        if self.spawn_heartbeat && self.options.heartbeat {
+            self.spawn_heartbeat = false;
```

Comment:
why set to `false`?

Reply:
It is `true`, it means it that we should spawn a heartbeat thread, otherwise a heartbeat thread has been spawned,